core/src/test/java/kafka/admin/DescribeAuthorizedOperationsTest.java

```diff
@@ -0,0 +1,275 @@
+        return List.of(
+            ClusterConfig.defaultBuilder()
+                .setTypes(Set.of(Type.KRAFT))
+                .setServerProperties(Map.of(GroupCoordinatorConfig.OFFSETS_TOPIC_PARTITIONS_CONFIG, "1"))
+                .setServerProperties(Map.of(GroupCoordinatorConfig.OFFSETS_TOPIC_REPLICATION_FACTOR_CONFIG, "1"))
+                .setBrokerSecurityProtocol(SecurityProtocol.SASL_PLAINTEXT)
+                .setControllerSecurityProtocol(SecurityProtocol.SASL_PLAINTEXT)
+                .build()
+        );
+    }
+
+    private static AccessControlEntry createAccessControlEntry(String username, AclOperation operation) {
+        return new AccessControlEntry(
+            new KafkaPrincipal(KafkaPrincipal.USER_TYPE, username).toString(),
+            AclEntry.WILDCARD_HOST,
+            operation,
+            ALLOW
+        );
+    }
+
+    private Map<String, Object> createAdminConfig(String username, String password) {
+        return createAdminConfig(username, password, Map.of());
+    }
+
+    private Map<String, Object> createAdminConfig(String username, String password, Map<String, Object> additionalConfigs) {
+        Map<String, Object> configs = new HashMap<>();
+        configs.put(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG, SecurityProtocol.SASL_PLAINTEXT.name);
+        configs.put(SaslConfigs.SASL_MECHANISM, "PLAIN");
+        configs.put(SaslConfigs.SASL_JAAS_CONFIG,
+            String.format("org.apache.kafka.common.security.plain.PlainLoginModule required username=\"%s\" password=\"%s\";", username, password));
+        configs.putAll(additionalConfigs);
+        return configs;
+    }
+
+    private void setupSecurity(ClusterInstance clusterInstance) {
+        try (Admin admin = clusterInstance.admin(createAdminConfig(JaasUtils.KAFKA_PLAIN_ADMIN, JaasUtils.KAFKA_PLAIN_ADMIN_PASSWORD))) {
+            ResourcePattern topicResource = new ResourcePattern(ResourceType.TOPIC, AclEntry.WILDCARD_RESOURCE, PatternType.LITERAL);
+
+            CreateAclsResult result = admin.createAcls(List.of(
+                new AclBinding(CLUSTER_PATTERN, ALTER_ENTRY),
+                new AclBinding(topicResource, DESCRIBE_ENTRY)
+            ));
+            assertDoesNotThrow(result::all);
+
+            assertDoesNotThrow(() -> clusterInstance.waitAcls(new AclBindingFilter(CLUSTER_PATTERN.toFilter(), ANY), Set.of(ALTER_ENTRY)));
+            assertDoesNotThrow(() -> clusterInstance.waitAcls(new AclBindingFilter(topicResource.toFilter(), ANY), Set.of(DESCRIBE_ENTRY)));
+        }
+    }
+
+    @ClusterTemplate("generator")
+    public void testConsumerGroupAuthorizedOperations(ClusterInstance clusterInstance) {
+        setupSecurity(clusterInstance);
+        try (Admin admin = clusterInstance.admin(createAdminConfig(JaasUtils.KAFKA_PLAIN_ADMIN, JaasUtils.KAFKA_PLAIN_ADMIN_PASSWORD))) {
+            assertDoesNotThrow(() -> admin.createTopics(List.of(new NewTopic("topic1", 1, (short) 1))));
```

Comment:
Do we need these `assertDoesNotThrow()` calls everywhere? If an exception was thrown wouldn't it automatically fail the test anyway?

Reply:
Removed useless `assertDoesNotThrow`. Thanks.

Reply:
There are still a bunch of `assertDoesNotThrow()` calls. Are they needed?

Reply:
Removed it. Thanks.